test/core/client_channel/resolvers/dns_resolver_test.cc

```diff
@@ -16,23 +16,46 @@
  *
```

Comment:
After talking to Yuchen, I've learnt that `resolve_address_test.cc` is a much better place for testing. I'll port these changes there tomorrow. The idea will the the same though: a chain of callbacks that will verify the behavior.